packages/internal/src/generate/graphqlCodeGen.ts

```diff
@@ -17,18 +17,25 @@ import type { LoadTypedefsOptions } from '@graphql-tools/load'
 import execa from 'execa'
 import { DocumentNode } from 'graphql'
 
-import { getPaths } from '@redwoodjs/project-config'
+import { getPaths, getConfig } from '@redwoodjs/project-config'
 
 import { getTsConfigs } from '../project'
 
 import * as rwTypescriptResolvers from './plugins/rw-typescript-resolvers'
-
 enum CodegenSide {
   API,
   WEB,
 }
 
 export const generateTypeDefGraphQLApi = async () => {
+  const config = getConfig()
+  if (config.experimental.useSDLCodeGenForGraphQLTypes) {
+    const paths = getPaths()
+    const sdlCodegen = await import('@sdl-codegen/node')
+    const dtsFiles = sdlCodegen.runFullCodegen('redwood', { paths })
```

Comment:
All of the work happens in the sdl-codegen side, there's a good argument that this is flawed (because what if Redwood changes the shape of this paths object) but I copied the types over into the lib. 

I think it's reasonable that at some point I chop this down to just the types from Path which are used, and then you aren't accidentally constrained on the shape of the object for anything but the API side.